packages/salesforcedx-vscode-apex/test/jest/oas/externalServiceRegistrationManager.test.ts

```diff
@@ -93,20 +99,23 @@ describe('ExternalServiceRegistrationManager', () => {
       esrHandler['initialize'](true, processedOasResult, fullPath);
 
       expect(esrHandler['isESRDecomposed']).toBe(true);
-      expect(esrHandler['processedOasResult']).toBe(processedOasResult);
       expect(esrHandler['oasSpec']).toBe(processedOasResult.openAPIDoc);
       expect(esrHandler['overwrite']).toBe(false);
       expect(esrHandler['originalPath']).toBe(fullPath[0]);
       expect(esrHandler['newPath']).toBe(fullPath[1]);
     });
 
     it('should initialize with overwrite set to true when paths are the same', async () => {
-      await esrHandler['initialize'](false, processedOasResult, ['/path/to/file.xml', '/path/to/file.xml', true]);
+      await esrHandler['initialize'](false, processedOasResult, ['/path/to/file.xml', '/path/to/file.xml']);
       expect(esrHandler['overwrite']).toBe(true);
     });
+    it('should initialize with overwrite set to false when paths differ only by extension', async () => {
+      await esrHandler['initialize'](false, processedOasResult, ['/path/to/original.xml', '/path/to/new.pdf']);
```

Comment:
```suggestion
      await esrHandler['initialize'](false, processedOasResult, ['/path/to/file.xml', '/path/to/file.pdf']);
```

The paths were originally differing by more than just the extension.